### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by PiekarzKarolina**

### CODE REVIEW

File: gui/functionality/add_cluster.py [hunk 1]
```diff
@@ -54,16 +65,25 @@ def auth_method_toggled(self):
...
+        """ Setups the worker that allows to run the add_cluster functionality
+        in the parallel thread.
+        """
         worker = Worker(self.add_cluster)
         worker.signals.result.connect(self.handle_complete_add_cluster)
         worker.signals.error.connect(self.handle_error_add_cluster)
         self.parent.threadpool.start(worker)
 
     def handle_complete_add_cluster(self):
+        """ Handles the competion of adding cluster.
```
Comment (jchyzy) [inline gui/functionality/add_cluster.py]: Typo: _completion_

File: gui/functionality/add_cluster.py [hunk 2]
```diff
@@ -124,26 +146,34 @@ def add_cluster(self):
             save_environment()
 
     def open_actions_file_dialog(self):
+        """ Opens the winodw dialog that allows to select an actions file.
```
Comment (jchyzy) [inline gui/functionality/add_cluster.py]: Typo: _window_

File: gui/functionality/add_cluster.py [hunk 3]
```diff
@@ -124,26 +146,34 @@ def add_cluster(self):
...
     def clear_actions_file(self):
+        """ Removes the selection of actions file.
+        """
         self.actions_file_name = None
         self.ui.selected_file_path_browser.setText("")
         self.ui.selected_file_path_browser.setEnabled(False)
         self.ui.clear_actions_file_button.setEnabled(False)
 
     def add_key(self):
+        """ Opens the winow dialog that allows to select a private key file.
```
Comment (jchyzy) [inline gui/functionality/add_cluster.py]: Typo: _window_

File: gui/functionality/add_cluster.py [hunk 4]
```diff
@@ -124,26 +146,34 @@ def add_cluster(self):
...
+        """ Opens the winodw dialog that allows to select an actions file.
+        """
         self.actions_file_name, _ = QFileDialog.getOpenFileName()
         if self.actions_file_name:
             self.ui.selected_file_path_browser.setText(self.actions_file_name)
             self.ui.selected_file_path_browser.setEnabled(True)
             self.ui.clear_actions_file_button.setEnabled(True)
 
     def clear_actions_file(self):
+        """ Removes the selection of actions file.
```
Comment (jchyzy) [inline gui/functionality/add_cluster.py]: The actions file? I'm not sure

File: gui/functionality/add_cluster.py [hunk 5]
```diff
@@ -124,26 +146,34 @@ def add_cluster(self):
...
+        """ Opens the winow dialog that allows to select a private key file.
+        """
         self.key_path, _ = QFileDialog.getOpenFileName()
         if self.key_path:
             self.ui.selected_key_browser.setText(self.key_path)
             self.ui.selected_key_browser.setEnabled(True)
             self.ui.clear_key_button.setEnabled(True)
 
     def clear_key(self):
+        """ Removes the selection of private key file.
```
Comment (jchyzy) [inline gui/functionality/add_cluster.py]: The/a private key maybe?

File: gui/functionality/add_cluster.py (hunk 4 above)
Comment (PiekarzKarolina) [reply gui/functionality/add_cluster.py]: Why? The method name is clear_actions_file so it clearly removes an action file.

File: gui/functionality/adjust_timeouts.py [hunk 6]
```diff
@@ -35,6 +45,10 @@ def __init__(self, data_provider, parent=None):
         self.data_provider.remove_cluster_signal.connect(self.handle_cluster_list_modification)
 
     def refresh_timeouts(self, cluster_name):
+        """ Fetches and refershes the timeouts of particular cluster.
```
Comment (jchyzy) [inline gui/functionality/adjust_timeouts.py]: Typo: _refreshes_
A particular? Not sure

File: gui/functionality/idact_notebook.py [hunk 7]
```diff
@@ -64,22 +75,33 @@ def __init__(self, data_provider, parent=None):
...
+        """ Setups the worker that allows to run the deploy_notebook functionality
+        in the parallel thread.
+        """
         worker = Worker(self.deploy_notebook)
         worker.signals.result.connect(self.handle_complete_deploy_notebook)
         worker.signals.error.connect(self.handle_error_deploy_notebook)
         self.parent.threadpool.start(worker)
 
     def handle_complete_deploy_notebook(self):
+        """ Handles the competion of deploy of the notebook.
```
Comment (jchyzy) [inline gui/functionality/idact_notebook.py]: Typo: _completion_

File: gui/functionality/idact_notebook.py [hunk 8]
```diff
@@ -172,6 +194,9 @@ def deploy_notebook(self):
         return
 
     def get_prepared_native_args(self):
+        """ Fetches the saved native_args from the native_args.json file.
+        Convarts them into proper format.
```
Comment (jchyzy) [inline gui/functionality/idact_notebook.py]: Typo: _converts_

File: gui/functionality/idact_notebook.py [hunk 9]
```diff
@@ -211,6 +242,8 @@ def remove_arguments(self):
         self.edit_native_arguments_window.set_that_data_changed()
 
     def save_arguments(self):
+        """ Saves the edited native arguments.
```
Comment (jchyzy) [inline gui/functionality/idact_notebook.py]: Without the? Not sure

File: gui/functionality/idact_notebook.py [hunk 10]
```diff
@@ -251,12 +289,20 @@ def set_that_data_changed(self):
         self.data_changed = True
 
     def show_warning_window(self, event):
+        """ Shows warning window that some changes may not have been saved
```
Comment (jchyzy) [inline gui/functionality/idact_notebook.py]: Dot missing

File: gui/functionality/idact_notebook.py [hunk 11]
```diff
@@ -251,12 +289,20 @@ def set_that_data_changed(self):
...
+            :param event: close event
+        """
         close_window = self.yes_or_no_window.show_message(
             "Some changes may not have been saved. \nDo you want to quit anyway?")
         self.yes_or_no_window.box.setDefaultButton(QMessageBox.No)
         if not close_window:
             event.ignore()
 
     def closeEvent(self, QCloseEvent):
+        """ Overriden method on closing Edit native arguments window
```
Comment (jchyzy) [inline gui/functionality/idact_notebook.py]: Dot missing

File: gui/functionality/manage_jobs.py [hunk 12]
```diff
@@ -52,6 +66,8 @@ def concurrent_show_jobs(self):
         self.parent.threadpool.start(worker)
 
     def handle_complete_show_jobs(self, jobs):
+        """ Handles the competion of show jobs function.
```
Comment (jchyzy) [inline gui/functionality/manage_jobs.py]: Typo: _completion_

File: gui/functionality/manage_jobs.py [hunk 13]
```diff
@@ -99,11 +124,17 @@ def concurrent_cancel_job(self):
         self.parent.threadpool.start(worker)
 
     def handle_complete_cancel_job(self):
+        """ Handles the competion of cancel job function.
```
Comment (jchyzy) [inline gui/functionality/manage_jobs.py]: Typo: _completion_

File: gui/functionality/popup_window.py [hunk 14]
```diff
@@ -1,20 +1,41 @@
...
+    See: :class:`.ProgramInfoWindow`, :class:`.ShowLogsWindow`,
+    :class:`.YesOrNoWindow`, :class:`.ShowJobsWindow`,
+    :class:`.EditNativeArgumentsWindow`
+"""
 from enum import Enum
 from PyQt5.QtWidgets import QWidget, QMessageBox
 
 
 class WindowType(Enum):
+    """ Avialble popup window types.
```
Comment (jchyzy) [inline gui/functionality/popup_window.py]: Typo: _available_

File: gui/functionality/popup_window.py [hunk 15]
```diff
@@ -1,20 +1,41 @@
...
+
+        :attr:`.error`: Window that shows an error message.
+
+    """
     success = 1
     error = 2
 
 
 class PopUpWindow(QWidget):
+    """ Helper widget window for idact notebook native arguments.
```
Comment (jchyzy) [inline gui/functionality/popup_window.py]: I think that this one is not only for native arguments

File: gui/functionality/remove_cluster.py [hunk 16]
```diff
@@ -34,16 +44,25 @@ def __init__(self, data_provider, parent=None):
...
+        """ Setups the worker that allows to run the remove_cluster functionality
+        in the parallel thread.
+        """
         worker = Worker(self.remove_cluster)
         worker.signals.result.connect(self.handle_complete_remove_cluster)
         worker.signals.error.connect(self.handle_error_remove_cluster)
         self.parent.threadpool.start(worker)
 
     def handle_complete_remove_cluster(self):
+        """ Handles the competion of removing cluster.
```
Comment (jchyzy) [inline gui/functionality/remove_cluster.py]: Typo: _completion_

File: gui/helpers/native_saver.py [hunk 17]
```diff
@@ -1,15 +1,27 @@
...
+
+    Similar modules: class:`.DataProvider`, :class:`.ParameterSaver`,
+    :class:`.UiLoader`, :class:`.Worker`, :class:`.ConfigurationProvider`
+"""
 import json
 import os
 
 
 class NativeArgsSaver:
+    """ Manages the native arguments.
```
Comment (jchyzy) [inline gui/helpers/native_saver.py]: Without the? Not sure

File: gui/helpers/worker.py [hunk 18]
```diff
@@ -1,13 +1,28 @@
...
 from PyQt5.QtCore import QRunnable, QObject, pyqtSignal, pyqtSlot
 
 
 class WorkerSignals(QObject):
+    """ 
+        :attr:`.finished`: Signal used to inform about normal finish of the worker job.
+        :attr:`.error`: Signal used to inform about finish with error of the worker job.
+                        The object parametr is the Error instance.
+        :attr:`.result`: Signal used to inform about the result of the worker job.
+                         The object parametr is the result of the job.
```
Comment (jchyzy) [inline gui/helpers/worker.py]: Typo: _parameter_

File: gui/helpers/worker.py [hunk 19]
```diff
@@ -1,13 +1,28 @@
...
+    :class:`.NativeArgsSaver`, :class:`.UiLoader`, :class:`.ConfigurationProvider`
+"""
 from PyQt5.QtCore import QRunnable, QObject, pyqtSignal, pyqtSlot
 
 
 class WorkerSignals(QObject):
+    """ 
+        :attr:`.finished`: Signal used to inform about normal finish of the worker job.
+        :attr:`.error`: Signal used to inform about finish with error of the worker job.
+                        The object parametr is the Error instance.
```
Comment (jchyzy) [inline gui/helpers/worker.py]: Typo: _parameter_

File: gui/helpers/worker.py [hunk 20]
```diff
@@ -18,6 +33,9 @@ def __init__(self, fn, *args, **kwargs):
 
     @pyqtSlot()
     def run(self):
+        """ Runs the task and signals about trown error or given result
```
Comment (jchyzy) [inline gui/helpers/worker.py]: Typo: _thrown_

File: gui/functionality/add_cluster.py (hunk 1 above)
Comment (PiekarzKarolina) [reply gui/functionality/add_cluster.py]: fixed

File: gui/functionality/add_cluster.py (hunk 2 above)
Comment (PiekarzKarolina) [reply gui/functionality/add_cluster.py]: fixed

File: gui/functionality/add_cluster.py (hunk 3 above)
Comment (PiekarzKarolina) [reply gui/functionality/add_cluster.py]: fixed

File: gui/functionality/add_cluster.py (hunk 4 above)
Comment (PiekarzKarolina) [reply gui/functionality/add_cluster.py]: Me too, I added it

File: gui/functionality/add_cluster.py (hunk 5 above)
Comment (PiekarzKarolina) [reply gui/functionality/add_cluster.py]: fixed

File: gui/functionality/adjust_timeouts.py (hunk 6 above)
Comment (PiekarzKarolina) [reply gui/functionality/adjust_timeouts.py]: fixed

File: gui/functionality/idact_notebook.py (hunk 7 above)
Comment (PiekarzKarolina) [reply gui/functionality/idact_notebook.py]: fixed

File: gui/functionality/idact_notebook.py (hunk 8 above)
Comment (PiekarzKarolina) [reply gui/functionality/idact_notebook.py]: fixed

File: gui/functionality/idact_notebook.py (hunk 9 above)
Comment (PiekarzKarolina) [reply gui/functionality/idact_notebook.py]: fixed

File: gui/functionality/idact_notebook.py (hunk 10 above)
Comment (PiekarzKarolina) [reply gui/functionality/idact_notebook.py]: fixed

File: gui/functionality/idact_notebook.py (hunk 11 above)
Comment (PiekarzKarolina) [reply gui/functionality/idact_notebook.py]: fixed

File: gui/functionality/manage_jobs.py (hunk 12 above)
Comment (PiekarzKarolina) [reply gui/functionality/manage_jobs.py]: fixed

File: gui/functionality/manage_jobs.py (hunk 13 above)
Comment (PiekarzKarolina) [reply gui/functionality/manage_jobs.py]: fixed

File: gui/functionality/popup_window.py (hunk 14 above)
Comment (PiekarzKarolina) [reply gui/functionality/popup_window.py]: fixed

File: gui/functionality/popup_window.py (hunk 15 above)
Comment (PiekarzKarolina) [reply gui/functionality/popup_window.py]: yes, you are right, fixed

File: gui/functionality/remove_cluster.py (hunk 16 above)
Comment (PiekarzKarolina) [reply gui/functionality/remove_cluster.py]: fixed

File: gui/helpers/native_saver.py (hunk 17 above)
Comment (PiekarzKarolina) [reply gui/helpers/native_saver.py]: I'll leave it here with the 'the'

File: gui/helpers/worker.py (hunk 18 above)
Comment (PiekarzKarolina) [reply gui/helpers/worker.py]: fixed

File: gui/helpers/worker.py (hunk 19 above)
Comment (PiekarzKarolina) [reply gui/helpers/worker.py]: fixed

File: gui/helpers/worker.py (hunk 20 above)
Comment (PiekarzKarolina) [reply gui/helpers/worker.py]: fixed

**Pull request merged**